modules/util/sec_to_time.go

```diff
@@ -18,10 +18,22 @@ import (
 // 45677465s	-> 1 year 6 months
 func SecToTime(duration int64) string {
 	formattedTime := ""
-	years := duration / (3600 * 24 * 7 * 4 * 12)
-	months := (duration / (3600 * 24 * 30)) % 12
-	weeks := (duration / (3600 * 24 * 7)) % 4
-	days := (duration / (3600 * 24)) % 7
+
+	// The following four variables are calculated by taking
+	// into account the previously calculated variables, this avoids
+	// pitfalls when using remainders. As that could lead to incorrect
+	// results when the calculated number equals the quotient number.
+	remainingDays := duration / (60 * 60 * 24)
+	years := remainingDays / 365
+	remainingDays -= years * 365
+	months := remainingDays / 30
+	remainingDays -= months * 30
```

Comment:
We could get the correct _average_ month length like this:

```suggestion
	months := remainingDays * 12 / 365
	remainingDays -= months * 365 / 12
```

This would solve the 364-day issue.

Reply:
I'm fine both with the current and the proposed approach.